service/matching/backlog_manager_test.go

```diff
@@ -29,28 +29,64 @@ import (
 	"testing"
 	"time"
 
-	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
 	enumspb "go.temporal.io/api/enums/v1"
-	"go.temporal.io/server/api/matchingservicemock/v1"
 	persistencespb "go.temporal.io/server/api/persistence/v1"
 	"go.temporal.io/server/common/dynamicconfig"
-	"go.temporal.io/server/common/log"
 	"go.temporal.io/server/common/metrics"
 	"go.temporal.io/server/common/primitives/timestamp"
+	"go.temporal.io/server/common/testing/testlogger"
 	"go.temporal.io/server/common/tqid"
 	"go.uber.org/mock/gomock"
 )
 
-func TestReadLevelForAllExpiredTasksInBatch(t *testing.T) {
-	controller := gomock.NewController(t)
+type BacklogManagerTestSuite struct {
```

Comment:
I revamped this test suite to understand it better and make it easier to port (what's needed). Didn't quite get there yet, though.

Reply:
I ported some of the tests now; and skipped the ones that cannot be fixed easily.